staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -182,6 +196,92 @@ func (p *Patcher) buildMergePatch(original, modified, current []byte) ([]byte, e
 	return patch, nil
 }
 
+// gvkSupportsPatchOpenAPIV3 checks if a particular GVK supports the patch operation.
+// It returns an error if the OpenAPI V3 could not be downloaded.
+func (p *Patcher) gvkSupportsPatchOpenAPIV3(gvk schema.GroupVersionKind) (bool, error) {
+	// Bypassing root to save apiserver memory.
+	gvSpec, err := p.OpenAPIV3Root.GVSpec(schema.GroupVersion{
+		Group:   p.Mapping.GroupVersionKind.Group,
+		Version: p.Mapping.GroupVersionKind.Version,
+	})
+	if err != nil {
+		return false, err
+	}
+	// TODO: Better error checking
+	for _, path := range gvSpec.Paths.Paths {
+		if path.Patch != nil {
+			if gvkMatches(p.Mapping.GroupVersionKind, path.Patch.Extensions) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// TODO: GVK parser should be moved elsewhere
+// Is this duplicated somewhere?
+func gvkMatches(targetGVK schema.GroupVersionKind, ext spec.Extensions) bool {
```

Comment:
There's a strong possibility that Alex has already written a piece of code to do that, I also suspect maybe sean has written a layer that goes on top of the OpenAPI client to find the object directly.

Reply:
I saw Sean's snippet in query params verifier v3. It's really concise and I'm going to opt to duplicate it because patch requires checking for two different kinds of gvks. Schemas can have a list of gvks while the patch endpoint can only have a single gvk.